cdap-app-fabric/src/main/java/io/cdap/cdap/app/runtime/AbstractProgramRuntimeService.java

```diff
@@ -123,7 +123,12 @@ public final RuntimeInfo run(ProgramDescriptor programDescriptor, ProgramOptions
     ProgramId programId = programDescriptor.getProgramId();
     ProgramRunId programRunId = programId.run(runId);
     DelayedProgramController controller = new DelayedProgramController(programRunId);
-    RuntimeInfo runtimeInfo = createRuntimeInfo(controller, programId, () -> cleanUpTask.get().run());
+    RuntimeInfo runtimeInfo = createRuntimeInfo(controller, programId, () -> {
+      Runnable task = cleanUpTask.get();
```

Comment:
nit: IMO , would be cleaner to remove this line, and do `if (cleanUpTask.get() != null) `